source/extensions/formatter/cel/cel.cc

```diff
@@ -20,18 +20,23 @@ CELFormatter::CELFormatter(const ::Envoy::LocalInfo::LocalInfo& local_info,
                            Expr::BuilderInstanceSharedPtr expr_builder,
                            const cel::expr::Expr& input_expr, absl::optional<size_t>& max_length,
                            bool typed)
-    : local_info_(local_info), expr_builder_(expr_builder), parsed_expr_(input_expr),
-      max_length_(max_length), typed_(typed) {
-  compiled_expr_ = Expr::createExpression(expr_builder_->builder(), parsed_expr_);
-}
+    : local_info_(local_info), max_length_(max_length), compiled_expr_([&]() {
+        auto compiled_expr = Expr::CompiledExpression::Create(expr_builder, input_expr);
+        if (!compiled_expr.ok()) {
+          throw EnvoyException(
+              absl::StrCat("failed to create an expression: ", compiled_expr.status().message()));
+        }
+        return std::move(compiled_expr.value());
```

Comment:
Do we need to have an explicit `std::move(...)` here?

Reply:
Yeah, compiler fails to infer the move for some reason. I think it's because the parent `compiled_expr` is still in scope on exit.